### PR TITLE
More work on PXA lowering

### PR DESCRIPTION
**Pull request by flaub**

### CODE REVIEW

File: pmlc/conversion/tile_to_pxa/pass.cc [hunk 1]
```diff
@@ -182,6 +183,42 @@ struct ScalarConstantOpConversion : public OpConversionPattern<ew::ScalarConstan
...
+    return from;
+  }
+  if (auto intoFloatType = intoType.dyn_cast<FloatType>()) {
+    if (auto fromFloatType = fromType.dyn_cast<FloatType>()) {
+      if (fromFloatType.getWidth() < intoFloatType.getWidth()) {
+        // FPExtOp: FloatType -> wider FloatType
+        return rewriter.create<mlir::FPExtOp>(loc, from, intoType).getResult();
+      }
+      // FPTruncOp: FloatType -> narrower FloatType
+      return rewriter.create<mlir::FPTruncOp>(loc, from, intoType).getResult();
```
Comment (jbruestle) [inline pmlc/conversion/tile_to_pxa/pass.cc]: Technically there is an uncovered case here of fp16 to bf16, which I think we should ignore, but just as an FYI.

File: pmlc/conversion/tile_to_pxa/pass.cc [hunk 2]
```diff
@@ -234,39 +304,51 @@ struct FirstOperand {
   }
 };
 
+void promoteTypes(ArrayRef<Value> operands, llvm::SmallVectorImpl<Value>* into) {
+  // TODO: add cast op for each operand wherever appropriate
```
Comment (jbruestle) [inline pmlc/conversion/tile_to_pxa/pass.cc]: I presume this code is meant to eventually compute the common subtype + then cast...

Comment (flaub) [reply pmlc/conversion/tile_to_pxa/pass.cc]: Yep, will do in a follow up

File: pmlc/conversion/tile_to_pxa/pass.cc (hunk 1 above)
Comment (flaub) [reply pmlc/conversion/tile_to_pxa/pass.cc]: Also there's a missing case to go from integer to floating point.

**Pull request merged**